packages/wrangler/src/publish/publish.ts

```diff
@@ -642,8 +648,20 @@ See https://developers.cloudflare.com/workers/platform/compatibility-dates for m
 						logger.log("Worker PipelineHash: ", result.pipeline_hash);
 				}
 			} catch (err) {
-				if (errIsScriptSizeOrStartupErr(err)) {
+				if (errIsScriptSize(err)) {
 					printOffendingDependencies(dependencies);
+				} else if (errIsStartupErr(err)) {
+					const youFailed =
+						"Your worker failed validation because it exceeded startup limits.";
+					const heresWhy =
+						"To ensure fast responses, we place constraints on worker startup -- like how much CPU it can use, or how long it can take.";
+					const heresTheProblem =
+						"Your worker failed validation, which means it hit one of these startup limits.";
```

Comment:
```suggestion
						"Your Worker failed validation because it exceeded startup limits.";
					const heresWhy =
						"To ensure fast responses, we place constraints on Worker startup -- like how much CPU it can use, or how long it can take.";
					const heresTheProblem =
						"Your Worker failed validation, which means it hit one of these startup limits.";
```